### PR TITLE
Add user

### PR DESCRIPTION
**Pull request by Havock2926**
Checking errors

### CODE REVIEW

File: Data/InstructionsForAddingUser.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+﻿1) Create User controller - post method, takes user as request, returns response class
+2) Create user class - contains all fields
```
Comment (Tavisca-vvinod) [inline Data/InstructionsForAddingUser.txt]: delete this file

File: Data/ItemDataLayer.cs [hunk 2]
```diff
@@ -8,6 +8,7 @@ namespace Data
 {
     public class ItemDataLayer
     {
+        //public static bool AddUser(User user)
```
Comment (Tavisca-vvinod) [inline Data/ItemDataLayer.cs]: delete all comments

File: Entities/Student.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Entities
```
Comment (Tavisca-vvinod) [inline Entities/Student.cs]: remove this file

File: Data/AddedUsers.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+﻿
```
Comment (Tavisca-vvinod) [inline Data/AddedUsers.txt]: Rename the file to Users.txt

File: Entities/User.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Entities
+{
+    public class User
+    {
+        public string UserEmailId { get; set; }
+        public int UserAge { get; set; }
```
Comment (Tavisca-vvinod) [inline Entities/User.cs]: Just curious, what will you do with a user's age in our system?

File: Host/Controllers/ItemController.cs [hunk 6]
```diff
@@ -27,5 +27,12 @@ public GetItemsResponse GetItemByItemId(int id)
             var responseFromService = ItemService.GetItemById(id);
             return responseFromService;
         }
+
+        [HttpPost]
+        public string GetStudentString([FromBody] Student student)
+        {
```
Comment (Tavisca-vvinod) [inline Host/Controllers/ItemController.cs]: remove this

File: Services/ItemService.cs [hunk 7]
```diff
@@ -40,5 +42,10 @@ public static GetItemsResponse GetItemById(int id)
             response.Items.Add(item);
             return response;
         }
+
+        public static string ModifyStudentString(Student student)
+        {
+            return student.Name + "hello" + student.Age + " " + student.Grade;
```
Comment (Tavisca-vvinod) [inline Services/ItemService.cs]: remove this method

File: Services/ItemService.cs [hunk 8]
```diff
@@ -29,8 +29,10 @@ public static GetItemsResponse GetItemById(int id)
                 return response;
             }
 
+            //var isadded = UserDataLayer.AddUser(user);
```
Comment (Tavisca-vvinod) [inline Services/ItemService.cs]: remove this comment

File: Services/UserService.cs [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+using System.Collections.Generic;
+using System.Text;
+
+namespace Services
+{
+    public class UserService
+    {
+        public static Response AddUser(User user)
+        {
+            bool IsAdded = UserDataLayer.AddUser(user);
```
Comment (Tavisca-vvinod) [inline Services/UserService.cs]: where is validation?
what if i sent UserName ="" in request?
what if i sent age = -90 or 0?

There is a method String.IsNullOrEmpty(string_name) in C#.
Google about it. Make use of it.

**Review by Tavisca-vvinod — CHANGES_REQUESTED** [review]

File: Data/InstructionsForAddingUser.txt (hunk 1 above)
Comment (Havock2926) [reply Data/InstructionsForAddingUser.txt]: Oh yeah I forgot to remove it
I'll do that

File: Entities/User.cs (hunk 5 above)
Comment (Havock2926) [reply Entities/User.cs]: Ummmmm
I dont know
I just added it 
I can remove it

File: Services/UserService.cs (hunk 9 above)
Comment (Havock2926) [reply Services/UserService.cs]: Alright maam I'll take a look online and edit the file

File: Host/Controllers/ItemController.cs (hunk 6 above)
Comment (Havock2926) [reply Host/Controllers/ItemController.cs:33]: maam i didnt get why i have to remove this 
Isnt it essential

File: Data/UserDataLayer.cs [hunk 10]
```diff
@@ -8,14 +8,42 @@ namespace Data
 {
     public class UserDataLayer
     {
+        //1. GetUserFromFile() - only read and return the list of users
```
Comment (Tavisca-vvinod) [inline Data/UserDataLayer.cs]: remove these comments

File: Entities/User.cs [hunk 11]
```diff
@@ -6,9 +6,15 @@ namespace Entities
...
-        public string UserName { get; set; }
-        public string UserPhone { get; set; }
+        public string EmailId { get; set; }
+        public string Name { get; set; }
+        public string ID { get; set; }
+        public string Phone { get; set; }
+        public string Password { get; set; }
 
+        public static implicit operator string(User v)
+        {
```
Comment (Tavisca-vvinod) [inline Entities/User.cs]: not needed

**Review by Tavisca-vvinod — CHANGES_REQUESTED** [review]

**Review by Tavisca-vvinod — APPROVED** [review]

**Pull request merged**